sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -24,27 +24,14 @@ def _fetch_native_df(self, query: t.Union[exp.Expression, str]) -> DF:
         self.execute(query)
 
         try:
-            df = self.cursor.fetch_pandas_all()
+            return self.cursor.fetch_pandas_all()
         except NotSupportedError:
             # Sometimes Snowflake will not return results as an Arrow result and the fetch from
             # pandas will fail (Ex: `SHOW TERSE OBJECTS IN SCHEMA`). Therefore we manually convert
             # the result into a DataFrame when this happens.
             rows = self.cursor.fetchall()
```

Comment:
this looks like a hack as well, why can't we just do

rows.to_pandas() or something